lib/trino-hive-formats/src/main/java/io/trino/hive/formats/line/json/JsonDeserializer.java

```diff
@@ -312,7 +319,12 @@ public SmallintDecoder()
         void decodeValue(JsonParser parser, BlockBuilder builder)
                 throws IOException
         {
-            SMALLINT.writeLong(builder, parser.getShortValue());
+            try {
+                SMALLINT.writeLong(builder, parser.getShortValue());
```

Comment:
According to `JsonParser#getShortValue()` a `JsonParseException` will be thrown here if the value can't be decoded to `short`. Since the return type of `parser.getShortValue()` is being implicitly coerced to `long`, the `TrinoException` path shouldn't be possible since it just validates the passed `long` value is within the range of `short`. Same goes for the other methods here, I think.

Reply:
think it should be `JsonProcessingException` instead of `JsonParseException` since the get<Type>Value() methods can also throw an `InputCoercionException` when the value falls outside of the type's range. Both `InputCoercionException` and `JsonParseException` subclasss `JsonProcessingException`. Same for comment below.